plugin/plugin.go

```diff
@@ -543,16 +543,22 @@ func awsInstanceToHost(instance *ec2.Instance) (*pb.ListHostsResponseHost, error
 	// External ID is the instance ID.
 	result.ExternalId = aws.StringValue(instance.InstanceId)
 
-	// First IP address is always the private IP address field if it's
-	// populated
+	// First IP address/dns name are always the private fields if they
+	// are populated
 	if aws.StringValue(instance.PrivateIpAddress) != "" {
 		result.IpAddresses = append(result.IpAddresses, aws.StringValue(instance.PrivateIpAddress))
 	}
+	if aws.StringValue(instance.PrivateDnsName) != "" {
+		result.DnsNames = append(result.DnsNames, aws.StringValue(instance.PrivateDnsName))
```

Comment:
Will these ever be repeated between this top level instance.PrivateDnsName and the DNS name under an interface's private ip address?

Similarly, will ip addresses ever be repeated as well?  If it's possible for these values to be repeated in different places it would be good to deduplicate for data cleanliness and when we try to write this data out to the db.

Reply:
They won't be repeated. The interface loop checks these top-level fields to ensure they are not the same.

Reply:
I've moved the duplicate check to just a rudimentary check of the whole current set of IP addresses/dns names.